internal/formats/common/spdxhelpers/to_syft_model.go

```diff
@@ -0,0 +1,373 @@
+package spdxhelpers
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/spdx/tools-golang/spdx"
+
+	"github.com/anchore/packageurl-go"
+	"github.com/anchore/syft/internal/formats/spdx22json/model"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/linux"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+)
+
+func ToSyftModel(doc *spdx.Document2_2) (*sbom.SBOM, error) {
+	spdxIDMap := make(map[string]interface{})
+
+	collectSyftPackages(spdxIDMap, doc)
+
+	collectSyftFiles(spdxIDMap, doc)
+
+	catalog := pkg.NewCatalog()
+
+	for _, v := range spdxIDMap {
+		if p, ok := v.(*pkg.Package); ok {
+			catalog.Add(*p)
+		}
+	}
+
+	return &sbom.SBOM{
+		Artifacts: sbom.Artifacts{
+			PackageCatalog:    catalog,
+			LinuxDistribution: findSyftLinuxRelease(doc),
+		},
+		Relationships: toSyftRelationships(spdxIDMap, doc),
+	}, nil
+}
+
+func findLinuxReleaseByDocument(doc *spdx.Document2_2) *linux.Release {
```

Comment:
This pattern is not widely adopted and also is generally the image scanned rather than a distro. I'd be fine taking this out although it _does_ work in some very limited cases. Not enough cases to make it especially useful, though.